src/main/kotlin/com/github/avrokotlin/avro4k/encoder/AvroStructureEncoder.kt

```diff
@@ -0,0 +1,95 @@
+package com.github.avrokotlin.avro4k.encoder
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.CompositeEncoder
+import kotlinx.serialization.encoding.Encoder
+
+@ExperimentalSerializationApi
+abstract class AvroStructureEncoder : AbstractAvroEncoder(), CompositeEncoder {
+    open fun shouldEncodeElement(descriptor: SerialDescriptor, index: Int): Boolean =
+            true
+
+    //region composite methods
+    final override fun encodeBooleanElement(descriptor: SerialDescriptor, index: Int, value: Boolean) {
+        if (shouldEncodeElement(descriptor, index)) {
+            resolveElementSchema(descriptor, index, false)
+            encodeBoolean(value)
+        }
+    }
```

Comment:
Now, for each element, we resolve the schema if it's a union. This permits to encode a string even if the union is `["null", "string", "int", {complex object I don't know why...}]`. Great for generic schemas but specific classes, for free